src/main/resources/interface-specification.yml

```
@@ -451,6 +451,114 @@ paths:
           schema:
             type: boolean
             default: false
+  '/agreements/filter/producers':
+      parameters:
+        - $ref: '#/components/parameters/CorrelationIdHeader'
+        - $ref: '#/components/parameters/IpAddress'
+      get:
+        tags:
+          - agreement
+        summary: Retrieves Producers from Agreements
+        description: Retrieves Producers from Agreements filtered by EService Name
+        operationId: getAgreementProducers
+        parameters:
+          - in: query
+            name: q
```

we can leave the `q` parameter in the BFF, but here we could use a more meaningful name
```suggestion
            name: eserviceName
```